pkg/envoy/lds/gateway.go

```diff
@@ -1,90 +1,89 @@
 package lds
 
 import (
-	"fmt"
-
+	xds_core "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
 	xds_listener "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
+	xds_tcp_proxy "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/tcp_proxy/v3"
 	"github.com/envoyproxy/go-control-plane/pkg/cache/types"
 	"github.com/envoyproxy/go-control-plane/pkg/wellknown"
-	"google.golang.org/protobuf/types/known/wrapperspb"
+	"github.com/golang/protobuf/ptypes"
 
 	"github.com/openservicemesh/osm/pkg/constants"
 	"github.com/openservicemesh/osm/pkg/envoy"
-	"github.com/openservicemesh/osm/pkg/errcode"
-	"github.com/openservicemesh/osm/pkg/service"
+	"github.com/openservicemesh/osm/pkg/identity"
+)
+
+const (
+	inboundMulticlusterGatewayFilterChainName = "inbound-multicluster-gateway-filter-chain"
 )
 
 func (lb *listenerBuilder) buildGatewayListeners() []types.Resource {
 	if !lb.cfg.GetFeatureFlags().EnableMulticlusterMode {
 		return nil
 	}
+
+	// TODO(draychev): what should this be?
+	svcIdent := identity.ServiceIdentity("osm-gateway.osm-system")
```

Comment:
Why is this required? This should already be set as a part of the `listenerBuilder` struct.

Reply:
I'm patching this and handing it off to @snehachhabria.  I can't remember why we need it here (bringing changes from a branch from December 2020).

Reply:
Per @shashankram - used https://github.com/openservicemesh/osm/blob/main/cmd/osm-controller/gateway.go#L44